http-server/src/main/java/org/triplea/server/moderator/toolbox/api/key/KeyHasher.java

```diff
@@ -0,0 +1,27 @@
+package org.triplea.server.moderator.toolbox.api.key;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Utility class to provide hashing functions for moderator API key.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class KeyHasher {
```

Comment:
This logic used to live in a controller factory. It was extracted here and a the key+password and key only variants were created.